### PR TITLE
[nnx] cleanup CallableProxy

### PR DESCRIPTION
**Pull request by cgarciae**
# What does this PR do?

* `CallableProxy` now create a `DelayedAccessor` by default if no accessor is given.
* Removes the creation of `DelayedAccessors` at many callsites `CallableProxy` is used.

### CODE REVIEW

File: flax/experimental/nnx/nnx/graph_utils.py [hunk 1]
```diff
@@ -330,7 +330,7 @@ def apply(
     accessesor = DelayedAccessor()
 
     def _apply(
-      accessesor, *args, **kwargs
+      accessesor: DelayedAccessor, *args, **kwargs
```
Comment (chiamp) [inline flax/experimental/nnx/nnx/graph_utils.py:333]: should `accessesor` be `accessor`?

File: flax/experimental/nnx/nnx/module.py [hunk 2]
```diff
@@ -283,21 +283,19 @@ def init(cls: type[M], *args, **kwargs) -> tuple[State, GraphDef[M]]:
...
   def create_abstract(cls: type[M]) -> type[M]:
-    accessesor = DelayedAccessor()
-
     def lift_rngs(kwargs: dict[str, tp.Any]):
       if 'rngs' in kwargs and isinstance(kwargs['rngs'], Rngs):
         kwargs['rngs'] = kwargs['rngs'].copy()
       return kwargs
 
-    def _create_abstract(accessesor, *args, **kwargs):
+    def _create_abstract(accessesor: DelayedAccessor, *args, **kwargs):
```
Comment (chiamp) [inline flax/experimental/nnx/nnx/module.py:291]: same in this file as well

File: flax/experimental/nnx/nnx/proxy_caller.py [hunk 3]
```diff
@@ -60,6 +38,28 @@ def __getitem__(self, key):
...
+  def __call__(self, accessor: DelayedAccessor, /, *args, **kwargs) -> tp.Any:
+    ...
+
+
+class CallableProxy:
+  def __init__(
+    self, callable: _AccessorCall, accessor: DelayedAccessor | None = None
+  ):
+    self._callable = callable
+    self._accesor = DelayedAccessor() if accessor is None else accessor
```
Comment (chiamp) [inline flax/experimental/nnx/nnx/proxy_caller.py:51]: should `self._accesor` be `self.accessor`?

File: flax/experimental/nnx/nnx/transforms.py [hunk 4]
```diff
@@ -92,11 +92,11 @@ def __call__(self, *args, **kwargs) -> tp.Any:
   def call(self) -> tp.Any:
     module = self
 
-    def check_and_call(*args, **kwargs):
+    def check_and_call(accesor: DelayedAccessor, *args, **kwargs):
```
Comment (chiamp) [inline flax/experimental/nnx/nnx/transforms.py:95]: Same in this file as well

**Review by chiamp — APPROVED** [review]

**Comment by chiamp** [comment]
There is also a CI failure

**Review by chiamp — CHANGES_REQUESTED** [review]
There is also a CI failure

**Comment by cgarciae** [comment]
Thanks! Fixed.

**Review by chiamp — APPROVED** [review]

File: flax/experimental/nnx/nnx/graph_utils.py [hunk 5]
```diff
@@ -330,10 +330,10 @@ def apply(
     accessesor = DelayedAccessor()
```
Comment (chiamp) [inline flax/experimental/nnx/nnx/graph_utils.py:330]: still some typos here: `accessesor -> accessor`

File: flax/experimental/nnx/nnx/module.py [hunk 6]
```diff
@@ -404,15 +402,15 @@ def pop(
...
   @property
   def apply(self: M) -> ApplyCaller[M]:
-    accessesor = DelayedAccessor()
-
-    def _apply(accessesor, *args, **kwargs) -> tuple[tp.Any, M]:
+    def _apply(
+      accessesor: DelayedAccessor, *args, **kwargs
+    ) -> tuple[tp.Any, M]:
       module = self.clone()
       fn = accessesor(module)
```
Comment (chiamp) [inline flax/experimental/nnx/nnx/module.py:409]: and here

**Pull request merged**